### PR TITLE
Let debug node show status indpendent of main output

### PR DESCRIPTION
**Pull request by dceejay**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Proposed changes

The PR lets the Debug node set it's status (under the node) independent of the main side bar and console outputs. It can select a msg.property or use a JSONata expression
image

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

**Comment by knolleary** [comment]
If a user wants the status and main output to match, they now have two properties to keep in sync.

How about adding a default option to the TypedInput that says `Use main output value` or similar. It will then behave as it does today without the user needing to edit the new field.

**Comment by dceejay** [comment]
if you untick it and re-tick it it should snap back to match

**Comment by knolleary** [comment]
An untick/tick of the status option resets it to `msg.payload` regardless of what's in the other box.

Still slightly concerned that's still an extra step needed - even if it is easier than editing the input box.

> And we don't want it to match if they select whole msg for the other.

True.

**Comment by dceejay** [comment]
will look at it

**Comment by dceejay** [comment]
Ok - fixed the field so they sync properly. 
re other thought(ie having an option in typed input) - can It have dynamic number of types ? ie not show the "same as above main option" (not real name)  if top one is full msg ?

**Comment by dceejay** [comment]
Now with automatic option (as default) - please suggest better name/label - that is basically the old mode where it shows whatever is in the main output - but now if it spots msg.status or msg.error then is uses those to set the status automatically, and if in full msg mode just tries to show the msg.payload.
image
image

File: packages/node_modules/@node-red/nodes/core/common/21-debug.js [hunk 1]
```diff
@@ -54,34 +38,90 @@ module.exports = function(RED) {
...
+            }
+        }
+
         this.on("input", function(msg, send, done) {
+            if (node.tostatus === true) {
+                prepareStatus(msg, function(err,debugMsg) {
+                    if (err) { node.error(err); return; }
+                    var output = debugMsg.msg;
+                    var st = (typeof output === 'string') ? output : util.inspect(output);
+                    if (st.length > 32) { st = st.substr(0,32) + "..."; }
```
[the remark this answers is not in the document]
Comment (dceejay) [reply packages/node_modules/@node-red/nodes/core/common/21-debug.js]: Yes - was thinking that incoming status messages would already be ok... which they would be... but errors wouldn't .

**Comment by dxdc** [comment]
@dceejay @knolleary I was quite confused by this behavior when installing the new NR (1.1.0).

- There are now two duplicate status options that appear now. There is a status 'check box' and a status 'dropdown'.
- However, if a user unchecks the status option, the status field goes away.

Is the actual behavior that if anything except `Automatic` is selected in the `status` dropdown, that it would *overwrite* the status field output with whatever the use defines?

If so, why not consolidate all of this to a single dropdown? It would work as follows:

- Add `None` and `Output` to the top of the dropdown
- Remove the `Automatic` option
- Remove the `status` checkbox
- Capitalize the word `Status` to match the rest of the form

Now, the user behavior would be as follows. The default would be `None` (same as the unticked status box), `Output` would reflect ticking the checkbox (historical behavior), and the remaining options would then override the output.

That is, unless, I'm missing the behavior? I do find it confusing though that there are two status options needed. It's also not clear to me how the below would be interpreted - e.g., would it be interpreted as a full status configuration vs. just a standard object, e.g.

```js
{
    fill: 'blue',
    shape: 'dot',
    text: 'Status message'
}
```

Lastly, for historical entries:
- Historical entries had a duplicated JSONata expression in both the output and the status fields. I didn't really understand why this happened.

img width="488" alt="Screen Shot 2020-07-01 at 8 24 19 AM" src="https://user-images.githubusercontent.com/7200365/86249001-49fae000-bb74-11ea-919a-4ffd3e3c25a2.png">

**Comment by dceejay** [comment]
Well (to me) the checkbox is consistent with the other two in that it enables and disables the output as the others do for their types of output. If enabled the extra option for the status then appears. (We could maybe lose the label - but it is about status :-)  

Having extra options appear as necessary when something is selected  is already  widely used - so I'm not sure why that is confusing in this instance.

The reason for the automatic option is that is is not quite the same behaviour as previously - in that it tries to do the best it can... previously if you set the main output to be complete msg we did not allow any status as we didn't know what to show. Now we default to showing msg.payload - (as was requested by a user) - and if connected to a status node or catch node it tries to show the relevant error message - in all other cases it shows what is selected as he main output. So I thought automatic was a good catchall term for that. I suppose output may also cover it, but to me doesn't indicate the slightly dynamic nature.

**Comment by dxdc** [comment]
> Having extra options appear as necessary when something is selected is already widely used - so I'm not sure why that is confusing in this instance.

Well, that's what I meant about just adding them to the dropdown. Otherwise there is a checkbox followed by a dropdown to define the behavior. But, I

Part of it may be how the existing nodes migrated then. I find this type of thing unhelpful... all my jsonata expressions were duplicated. It should migrate using "Automatic" mode in these cases (I think?). The previous image I showed is after I edited it.

So, for my understanding automatic will distinguish between msg.payload, an error message, and (?) maybe also this?

```js
{
    fill: 'blue',
    shape: 'dot',
    text: 'Status message'
}
```

image

**Comment by dceejay** [comment]
Ah right - so yes - it would probably make more sense if they migrated to automatic mode as that is most similar to previous behaviour - and would losing the label help ?

Still not sure what you mean by that example status json ? where is that coming from ?

**Comment by dxdc** [comment]
> Ah right - so yes - it would probably make more sense if they migrated to automatic mode as that is most similar to previous behaviour - and would losing the label help ?

It might. Thinking about this some more, I think the problem I'm having is that the UI indicates that Output is rerouted to... and then the user can choose a number of options. If the user chooses status among those, some of the items in the dropdown indicate mixed behavior. 

For example... in the above case, will the Output get routed to status, but then, immediately replaced by the output of another JSONata expression?

I didn't dig through the logic within the code, but if this is truly an "override", then maybe it should be more clearly stated that it will override the status output with something else? (Either that, or distinguish maybe differently from the "Automatic" label?)

> Still not sure what you mean by that example status json ? where is that coming from ?

Sorry, I meant, is it possible to change the ring/color of the status for the debug node? Previously, this was impossible I think since it was just routing the text.

**Comment by dceejay** [comment]
hmm - yes - now that you put it like that - I guess it is an override or alternative. Will think on.

Re the example - no - the status only show the text of the property value - it doesn't try to inspect it to see if it's actually a status type object with text, fill and shape

**Comment by dceejay** [comment]
maybe renaming the label fro automatic to show Output would help ? (and fix the default migration to be to this mode) 
image

**Comment by dxdc** [comment]
Much cleaner! Yes... I like it.

What about just `default` or `default output`?

**Comment by dxdc** [comment]
> Re the example - no - the status only show the text of the property value - it doesn't try to inspect it to see if it's actually a status type object with text, fill and shape

This would be a cool addition at some point.

**Comment by dceejay** [comment]
thinking some more about - (now that I understand the request) - it actually makes more sense than "auto" mode... so will try this and see.

**Comment by dxdc** [comment]
@dceejay awesome job with this! loving the new display in 1.1.1.

**Pull request merged**